src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -1361,6 +1345,25 @@ DistributedExecutionRequiresRollback(List *taskList)
 
 	Task *task = (Task *) linitial(taskList);
 
+	if (taskCount == 1 && TaskAccessesOnlyCitusLocalTables(task) &&
```

Comment:
I'm really not sure whether we should fix this issue or not as on MX these types of tx blocks would start to fail whatever we do. @marcocitus what would be your take on this?

Reply:
Similarly, we have `TaskAccessesOnlyCitusLocalTables ` just because we wouldn't want to break users that works on single node and not works when scale out such as:
```SQL
BEGIN;
INSERT INTO dist SELECT * FROM dist;
PREPARE ...
```